sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/QueryRewrites.java

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.search.documents.models;
+
+/**
+ * Configuration for how semantic search rewrites a query.
+ */
+public class QueryRewrites {
+
+    private final QueryRewritesType rewritesType;
+    private Integer count;
+
+    /**
+     * Creates a new instance of {@link QueryRewrites}.
+     *
+     * @param rewritesType The type of query rewrites to perform.
+     */
+    public QueryRewrites(QueryRewritesType rewritesType) {
+        this.rewritesType = rewritesType;
```

Comment:
Should make this required as we did with Semantic Answers and Captions.

```suggestion
        this.rewritesType = Objects.requireNonNull(rewritesType, "'rewritesType' cannot be null");
```

Will also need to add a `@throws` into the Javadoc.